src/app/weather-service.service.ts

```diff
@@ -0,0 +1,29 @@
+import {Injectable} from '@angular/core';
+import {HttpClient, HttpHeaders} from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class WeatherServiceService {
+  locationLatLon = [`latitude=53.105&longitude=-3.913`,`latitude=53.317&longitude=-3.482`,`latitude=52.132&longitude=-4.547`];
+  i:any
+  private baseUrl = `https://api-metoffice.apiconnect.ibmcloud.com/v0/forecasts/point/hourly?excludeParameterMetadata=true&includeLocationName=true&`;
+
+  constructor(private http: HttpClient) { }
+
+    getLocationData(id:number) {
+    const apiUrl = `${this.baseUrl + this.locationLatLon[id]}`;
+    let headers = new HttpHeaders();
+    headers = headers.append('X-IBM-Client-Id', 'da539e1ef8b771d86e1aa2cf44b8efb8');
+    headers = headers.append('X-IBM-Client-Secret', '93f9d7ea7fb3e2e6b1d3f770ce553405');
```

Comment:
we never check in secrets like this, as if someone malitious got hold of them they could spam the API using your details

might be worth doing something similar to hazard manager, where it'll read in a config file on startup to get these dynamically & that way they could be removed from source control (I can show you how that works at some point as i'm not sure if there are good docs on it online)